### PR TITLE
Add NumberFormatException when the port isn't correctly formatted for Firefox, Edge and Chrome Drivers

### PR DESCRIPTION
**Pull request by MustafaAgamy**
### Description
Add **"NumberFormatException"** when the port isn't correctly formatted for **Firefox**, **Edge** and **Chrome** Drivers at:
**- GeckoDriverService
- EdgeDriverService
- ChromeDriverService**

### Motivation and Context
Due to Operating System langauge set in Arabic, middle east users were getting **"SessionNotCreatedException"** which wasn't detailed enough, now the users will get a detailed **"NumberFormatException"** with the recommended solution and a link to the docs for more info
*Please note that I already had tests created for previous PR, now I just adjusted those Tests to work better with the given scenario (Expected to Throw NumberFormatException)
### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
Bug fix, Enhancement

___

### **Description**
- Added proper error handling for non-English system locales in Chrome, Edge, and Firefox WebDrivers
- Throws NumberFormatException with detailed message when system locale is Arabic
- Added validation check for port number formatting in all three driver services
- Updated test cases to verify the new NumberFormatException behavior
- Includes helpful error message with instructions to set JVM language arguments
- Added documentation link in error message for more information

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Error handling</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ChromeDriverService.java</strong><dd><code>Add locale validation for Chrome driver port formatting</code>&nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/chrome/ChromeDriverService.java

<li>Added check for non-English system locale when formatting port numbers<br> <li> Throws NumberFormatException with detailed message for Arabic locale<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14946/files#diff-405c68a6b743ea8f4b340b5ea0c92c802a24109b4c38818335f6ca39511e2bc8">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>EdgeDriverService.java</strong><dd><code>Add locale validation for Edge driver port formatting</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/edge/EdgeDriverService.java

<li>Added check for non-English system locale when formatting port numbers<br> <li> Throws NumberFormatException with detailed message for Arabic locale<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14946/files#diff-62903f54af704e41bd4657395512ce9c62fa0d4966ef3f0f665fcc045e7b15c4">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>GeckoDriverService.java</strong><dd><code>Add locale validation for Firefox driver port formatting</code>&nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/firefox/GeckoDriverService.java

<li>Added import for java.util.Locale<br> <li> Added check for non-English system locale when formatting port numbers<br> <li> Throws NumberFormatException with detailed message for Arabic locale<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14946/files#diff-a441a4f6b8b9cfb6e4e25122a31c17dbf86beae9bf02521e1516b27fcd130245">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ChromeDriverFunctionalTest.java</strong><dd><code>Update Chrome driver tests for locale validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/chrome/ChromeDriverFunctionalTest.java

<li>Renamed test method to shouldThrowNumberFormatException<br> <li> Updated test to verify NumberFormatException is thrown for Arabic <br>locale<br> <li> Removed successful launch test case<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14946/files#diff-0bc642dab2d557261d4507422be800146f1ff78a1889f6f728e7eae08a76953f">+7/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>EdgeDriverFunctionalTest.java</strong><dd><code>Update Edge driver tests for locale validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/edge/EdgeDriverFunctionalTest.java

<li>Renamed test method to shouldThrowNumberFormatException<br> <li> Updated test to verify NumberFormatException is thrown for Arabic <br>locale<br> <li> Removed successful launch test case<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14946/files#diff-9b41e954ffa07a702247ef4ea6b0ca2d9b2a4eca9cf9686c8b2416550e29db83">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>FirefoxDriverTest.java</strong><dd><code>Update Firefox driver tests for locale validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/firefox/FirefoxDriverTest.java

<li>Renamed test method to shouldThrowNumberFormatException<br> <li> Updated test to verify NumberFormatException is thrown for Arabic <br>locale<br> <li> Removed successful launch test case<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14946/files#diff-6ff6428e62eafbf44f2cee9400a71f0a7f3276807d4b611000b7434c582752a3">+7/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by MustafaAgamy** [comment]
@pujagani FYI

**Comment by Delta456** [comment]
I wonder if can use this string as a const and use that everywhere. I feel that would be a good idea.

File: java/src/org/openqa/selenium/chrome/ChromeDriverService.java [hunk 1]
```diff
@@ -285,6 +285,10 @@ protected void loadSystemProperties() {
     protected List<String> createArgs() {
       List<String> args = new ArrayList<>();
       args.add(String.format("--port=%d", getPort()));
+      if(Locale.getDefault(Locale.Category.FORMAT).getLanguage().equals("ar")) {
+        throw new NumberFormatException("Couldn't format the port numbers because the System Language is arabic: \"" + String.format("--port=%d", getPort()) +
+        "\", please make sure to add the required arguments \"-Duser.language=en -Duser.region=US\" to your JVM, for more info please visit :" + "\n  https://www.selenium.dev/documentation/webdriver/browsers/");
+      }
```
Comment (Delta456) [inline java/src/org/openqa/selenium/chrome/ChromeDriverService.java]: I did prefer 

```java
String.format("Couldn't format the port numbers because the System Language is arabic: \"--port=%d"\", please make sure to add the required arguments \"-Duser.language=en -Duser.region=US\" to your JVM, for more info please visit :\n  https://www.selenium.dev/documentation/webdriver/browsers/",  getPort());
```

Comment (MustafaAgamy) [reply java/src/org/openqa/selenium/chrome/ChromeDriverService.java]: Changes done

**Comment by MustafaAgamy** [comment]
@pujagani
@diemol
Bump

**Review by diemol — COMMENTED** [review]
Can't this be placed in the super class?

**Comment by MustafaAgamy** [comment]
> Can't this be placed in the super class?

@diemol  Done, please check

Also added Tests for Safari and InternetExplorer alongside the added tests for Firefox, Chrome and Edge

**Review by pujagani — APPROVED** [review]
LGTM. Thank you @MustafaAgamy!

**Comment by MustafaAgamy** [comment]
> LGTM. Thank you @MustafaAgamy!

@pujagani  
I fixed the formatting issues related to the class files,

The test failures aren't related to the fix 

Thank you

**Comment by MustafaAgamy** [comment]
@pujagani 

Sorry one more time, I forgot another line fix my bad

**Comment by MustafaAgamy** [comment]
@pujagani 
can you please re-reun the PR tests?

And I see tests failure related to something else, not sure what's the root cause

thank you

File: .idea/misc.xml [hunk 2]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="ExternalStorageConfigurationManager" enabled="true" />
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
   </component>
```
Comment (diemol) [inline .idea/misc.xml]: Please avoid changes to this file.

File: .idea/kotlinc.xml
Comment (diemol) [inline .idea/kotlinc.xml:1]: Please do not remove this file.

**Review by diemol — CHANGES_REQUESTED** [review]

**Comment by MustafaAgamy** [comment]
@diemol @pujagani  done changes reverted

Please check

**Comment by MustafaAgamy** [comment]
@diemol  Bump

**Comment by MustafaAgamy** [comment]
@pujagani
@diemol

Test failures not related to the change so everything is good, I guess

**Comment by pujagani** [comment]
Thank you! @MustafaAgamy
